pkg/engine2/path_selection/path_expansion.go

```diff
@@ -437,11 +437,18 @@ func expandPath(
 			return
 		}
 
-		valid, err := checkUniquenessValidity(ctx, source.id, target.id)
-		if err != nil {
+		// if the edge doesnt exist in the actual graph we need to check uniqueness validity
+		_, err := ctx.RawView().Edge(source.id, target.id)
+		if errors.Is(err, graph.ErrEdgeNotFound) {
+			valid, err := checkUniquenessValidity(ctx, source.id, target.id)
+			if err != nil {
+				errs = errors.Join(errs, err)
```

Comment:
Does this need to return?

Reply:
valid should be false, but yeah not a bad idea to have it return anyways

Reply:
I guess not because if it returned an error, it should also return `!valid`